entities/entities/spawned_weapon/init.lua

```diff
@@ -52,26 +52,45 @@ function ENT:Use(activator, caller)
         return
     end
 
-    local ammoType = weapon:GetPrimaryAmmoType()
     local CanPickup = hook.Call("PlayerCanPickupWeapon", GAMEMODE, activator, weapon)
     local ShouldntContinue = hook.Call("PlayerPickupDarkRPWeapon", nil, activator, self, weapon)
     if not CanPickup or ShouldntContinue then return end
 
-    local newAmmo = activator:GetAmmoCount(ammoType) -- Store ammo count before weapon pickup
-
+    -- Store ammo count before weapon pickup
+    local primaryAmmoType = weapon:GetPrimaryAmmoType()
+    local secondaryAmmoType = weapon:GetSecondaryAmmoType()
     weapon:Remove()
 
-    activator:Give(class)
-
-    weapon = activator:GetWeapon(class)
-    newAmmo = newAmmo + (self.ammoadd or 0) -- Gets rid of any ammo given during weapon pickup
+    weapon = activator:Give(class)
+
+    local clip1, clip2 = self.clip1, self.clip2
+    if weapon:IsValid() then
```

Comment:
This will never be true because the weapon is removed 5 lines above this line

Reply:
weapon = activator:Give(class)